cpp/tests/test_compartments_simulation.cpp

```diff
@@ -55,6 +56,38 @@ TEST(TestCompartmentSimulation, integrator_uses_model_reference)
     ASSERT_NEAR(sim.get_result().get_last_value()[0], 3.0, 1e-5);
 }
 
+TEST(TestCompartmentSimulation, copy_simulation)
+{
+
+    auto sim = mio::Simulation<double, MockModel>(MockModel(), 0.0);
+
+    auto sim_copy_cnstr = mio::Simulation<double, MockModel>(sim);
```

Comment:
Not a bad test, but please add a brief comment explaining the test case.

The sim_copy_cnstr uses assignment as well. Try
```suggestion
    mio::Simulation<double, MockModel> sim_copy_cnstr(sim);
```